pvlib/ivtools.py

```diff
@@ -262,6 +264,177 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
                                      v_oc)
 
 
+def fit_sdm_desoto(celltype, v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc,
+                   cells_in_series, temp_ref=25, irrad_ref=1000):
+    """
+    Calculates the five parameters for the single diode equation using
```

Comment:
"Calculates the parameters for the De Soto single diode model using the procedure described in [1]".